module-utils/log/Logger.cpp

```diff
@@ -129,15 +152,22 @@ namespace Log
         loggerBufferCurrentPos = 0;
         addLogHeader(level, file, line, function);
 
-        const auto sizeLeft = loggerBufferSizeLeft();
-        const auto result   = vsnprintf(&loggerBuffer[loggerBufferCurrentPos], sizeLeft, fmt, args);
+        const auto sizeLeft   = loggerBufferSizeLeft();
+        const auto resultLog  = vsnprintf(&loggerBuffer[loggerBufferCurrentPos], sizeLeft, fmt, args);
+        const auto resultSize = sprintf(&loggerBuffer[loggerBufferCurrentPos + resultLog],
+                                        " [%d/%d B:%d]",
+                                        buffer.getCurrentBuffer()->getSize(),
+                                        buffer.getCurrentBuffer()->getCapacity(),
+                                        buffer.getCurrentIndex());
+        const auto result     = resultLog + resultSize;
         if (0 <= result) {
             const auto numOfBytesAddedToBuffer = static_cast<size_t>(result);
             loggerBufferCurrentPos += (numOfBytesAddedToBuffer < sizeLeft) ? numOfBytesAddedToBuffer : (sizeLeft - 1);
             loggerBufferCurrentPos += snprintf(&loggerBuffer[loggerBufferCurrentPos], loggerBufferSizeLeft(), "\n");
 
             logToDevice(Device::DEFAULT, loggerBuffer, loggerBufferCurrentPos);
-            circularBuffer.put(std::string(loggerBuffer, loggerBufferCurrentPos));
+            buffer.getCurrentBuffer()->put(std::string(loggerBuffer, loggerBufferCurrentPos));
+            checkBufferState();
```

Comment:
consider moving it out (if possible) to a helper function to not repeat the same code :)

Reply:
Done